src/Services/DatabaseAuditLog.cs

```diff
@@ -0,0 +1,38 @@
+using IdentityServerConfig.Data;
+using IdentityServerConfig.Models;
+using Microsoft.AspNetCore.Components.Authorization;
+
+namespace IdentityServerConfig.Services;
+
+public class DatabaseAuditLog : IDatabaseAuditLog
+{
+    private readonly AuditContext _auditContext;
+    private readonly AuthenticationStateProvider _authenticationStateProvider;
+    
+    public DatabaseAuditLog(AuditContext auditContext, AuthenticationStateProvider authenticationStateProvider)
+    {
+        _auditContext = auditContext;
+        _authenticationStateProvider = authenticationStateProvider;
+    }
+    
+    public int Log(string action)
+    {
+        var user = _authenticationStateProvider.GetAuthenticationStateAsync().Result.User;
+        var userId = user.Claims.FirstOrDefault(c => c.Type.Equals("http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier"))?.Value;
```

Comment:
I believe there are builtin constants for these claim types available within .NET which we should probably use here.

Reply:
Changed it in 27976c1